_posts/2021-11-09-logicdrop-automating-quarkus-with-gitlab.adoc

```diff
@@ -0,0 +1,249 @@
+
+image::gitlab/view-sentry.png[Sentry Errors,align="center"]
+
+=== Coverage and Quality Reports
+Jacoco generates coverage reports for each project. Metrics are maintained and visible throughout the lifetime of each service.
+
+image::gitlab/coverage.png[Coverage Report,align="center"]
+
+Additionally, Code Climate is used to measure changes in quality between the default branch and each merge request throughout the lifetime of every service.
+
+=== Distributed Service Traces
+Jaeger provides us with insights into how our APIs are used and lets us trace interactions between multiple services.
+
+image::gitlab/view-jaeger1.png[Jaeger Query,align="center"]
+image::gitlab/view-jaeger2.png[Jaeger Details,align="center"]
+
+TIP: Jaeger has been invaluable to us because we heavily rely on single-responsibility services that frequently communicate with other services.
+
+=== Dependency Changes
+Dependabot is wired in to let us know when there are changes to any dependencies.  
+
+image::gitlab/dependabot.png[Dependabot,align="center"]
+
+TIP: This is especially helpful when our dependencies get released and other projects need to be made aware of it. Since it creates a merge request, the tests are run and we can see if the upgrade will break anything.
+
+=== Environments and Pod Health
+Every environment is monitored and you can see what pods are active in them by adding a couple of GitLab-specific annotations to the Quarkus Kubernetes configuration.
+
+image::gitlab/view-pods.png[Pod Health,align="center"]
+
+=== Cluster Logs
+Kubernetes logs can be viewed right from GitLab, eliminating the need for access to the cluster by developers. 
+
+Logs can be viewed per environment or filtered for specific pods.
+
+image::gitlab/view-logs.png[Cluster Logs,align="center"]
+
+=== Prometheus Metrics
+Prometheus metrics are exposed using GitLab's monitoring and metrics. 
+
+image::gitlab/prom.png[Prometheus Metrics,align="center"]
+
+=== Value Stream Analytics
+We can use GitLab's https://docs.gitlab.com/ee/user/analytics/value_stream_analytics.html[Value Stream Analytics^] to track the velocity within any project. This gives us rough insights into the time it takes from the moment a ticket is created to when it is deployed.
+
+image::gitlab/valuestream.png[Value Stream Analytics,align="center"]
+
+== Conclusion
+Integrating Quarkus deeply into GitLab has added significant value to our process and was well worth the little effort it took. 
+
+Building upon our _level the playing field_ philosophy, developing, debugging, deploying, and monitoring cloud-native services is more optimal than ever.
+
+Because of the natural fit Quarkus has with cloud technologies, and the functionality provided through extensions, we have been able to create a comprehensive DevOps ecosystem that normally would be challenging to setup and orchestrate.
```

Comment:
```suggestion
Because of the natural fit Quarkus has with cloud technologies, and the functionality provided through extensions, we have been able to create a comprehensive DevOps ecosystem that normally would be challenging to set up and orchestrate.
```